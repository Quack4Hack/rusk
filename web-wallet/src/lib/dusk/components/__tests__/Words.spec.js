import {
	afterEach,
	describe,
	expect,
	it
} from "vitest";
import {
	cleanup,
	render
} from "@testing-library/svelte";

import { Words } from "..";

describe("Words", () => {
	afterEach(cleanup);

	it("should render the \"Words\" component with underscores for empty string", () => {
		const { container } = render(Words, { props: { words: ["", "", ""] } });

		expect(container.firstChild).toMatchSnapshot();
	});

	it("should render the \"Words\" component with the passed words", () => {
		const { container } = render(Words, { props: { words: ["snow", "winter", "christmas"] } });

		expect(container.firstChild).toMatchSnapshot();
	});
});
