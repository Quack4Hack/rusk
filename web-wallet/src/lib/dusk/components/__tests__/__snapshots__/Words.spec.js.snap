// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Words > should render the "Words" component with the passed words 1`] = `
<ol
  class="dusk-words"
>
  <li
    class="dusk-words__word"
  >
    snow
  </li>
  <li
    class="dusk-words__word"
  >
    winter
  </li>
  <li
    class="dusk-words__word"
  >
    christmas
  </li>
</ol>
`;

exports[`Words > should render the "Words" component with underscores for empty string 1`] = `
<ol
  class="dusk-words"
>
  <li
    class="dusk-words__word"
  >
    _____
  </li>
  <li
    class="dusk-words__word"
  >
    _____
  </li>
  <li
    class="dusk-words__word"
  >
    _____
  </li>
</ol>
`;
